wagtail/admin/widgets/button.py

```diff
@@ -65,12 +65,35 @@ def __init__(self, *args, **kwargs):
     def dropdown_buttons(self):
         raise NotImplementedError
 
-    def render(self):
-        return render_to_string(self.template_name, {
+    def get_context_data(self):
+        return {
             'buttons': self.dropdown_buttons,
             'label': self.label,
             'title': self.attrs.get('title'),
-            'is_parent': self.is_parent})
+            'is_parent': self.is_parent
+        }
+
+    def render(self):
+        return render_to_string(self.template_name, self.get_context_data())
+
+
+class ButtonWithDropdown(BaseDropdownMenuButton):
+    template_name = 'wagtailadmin/pages/listing/_button_with_dropdown.html'
+
+    def __init__(self, *args, **kwargs):
+        self.btn_classes = kwargs.pop('btn_classes', set())
+        self.buttons_data = kwargs.pop('buttons_data', [])
+        super().__init__(*args, **kwargs)
+
+    def get_context_data(self):
+        context = super().get_context_data()
+        context['btn_classes'] = self.btn_classes
```

Comment:
nitpick: although we use `.btn` in CSS, I feel this can be more explicit `button_classes`